### PR TITLE
nxcdb: refactor shared database/workspace setup code & allow for creation/setting of workspaces outside of nxcdb interactive console

### PR DESCRIPTION
**Pull request by Marshall-Hallenbeck**
- Refactors some database related code like creating the DB engine, creating workspaces, etc to be a single shared location
- Allows creation of workspaces via command line parameters in nxcdb
- Allows setting of active workspace in nxc config via command line parameters in nxcdb
- Remove redundant WS_PATH since there's already WORKSTATION_PATH in paths file

Now if you want to automate scanning, but have it in separate workspaces, and dont want to do some sed jankiness in the config, you can create a new workspace and make it the active one for NetExec with some command line calls.

### CODE REVIEW

**Review by NeffIsBack — CHANGES_REQUESTED** [review]
Rest looks good

**Comment by NeffIsBack** [comment]
image

LGTM

File: nxc/paths.py [hunk 1]
```diff
@@ -8,7 +8,7 @@
     TMP_PATH = os.getenv("LOCALAPPDATA") + "\\Temp\\nxc_hosted"
 if hasattr(sys, "getandroidapilevel"):
     TMP_PATH = os.path.join("/data", "data", "com.termux", "files", "usr", "tmp", "nxc_hosted")
-WS_PATH = os.path.join(NXC_PATH, "workspaces")
+
```
[the remark this answers is not in the document]
Comment (NeffIsBack) [reply nxc/paths.py:11]: Resolved it

**Review by NeffIsBack — CHANGES_REQUESTED** [review]
image

Ah one bug left. Now it tries to always create the default workspace on every command

**Comment by NeffIsBack** [comment]
Added some other features:

- New command to list all existing workspaces and print out the current active workspace
- Change the workspace listing output to match git output style

**Comment by Marshall-Hallenbeck** [comment]
Fixed the bug where it tries to create the default workspace every run. Ready for final review.

**Review by NeffIsBack — APPROVED** [review]
Working now, LGTM

**Pull request merged**